src/Jmikola/AutoLogin/Authentication/Provider/AutoLoginProvider.php

```diff
@@ -49,11 +47,13 @@ public function __construct(UserProviderInterface $userProvider, UserCheckerInte
     }
 
     /**
+     * @param TokenInterface $token
+     *
      * @see Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface::authenticate()
      */
     public function authenticate(TokenInterface $token)
     {
-        if (!$this->supports($token)) {
+        if (!$token instanceof AutoLoginToken || !$this->supports($token)) {
```

Comment:
I'll remove this before merging, as this check is already done in the `supports()` method.